### PR TITLE
RM4 support [WIP]

### PR DESCRIPTION
**Pull request by ynuyasha**
RM4 support [WIP]

Add support for rm4 devices described in https://github.com/mjg59/python-broadlink/blob/0.13.0/broadlink/__init__.py#L50

### CODE REVIEW

**Comment by ynuyasha** [comment]
can you try to build binding from my repo and try if works for your device?
I resolved some bugs on authentication

**Comment by tknaller** [comment]
just tried, it happens as I had the problem before:
```
08:37:02.135 [ERROR] [adlink.handler.BroadlinkRemoteHandler] - rm3:c8b2c2c2[?]: java.net.ProtocolException: Response from device is not valid. (0x22=0xF9,0x23=0xFF,0x24=0xFF)
08:37:02.138 [ERROR] [adlink.handler.BroadlinkRemoteHandler] - rm3:c8b2c2c2[?]: ANTOBUG: linea 268 Basethinghandler
08:37:02.139 [ERROR] [adlink.handler.BroadlinkRemoteHandler] - rm3:c8b2c2c2[?]: Attempting to authenticate prior to getting device status FAILED. Will mark as offline
08:37:02.140 [ERROR] [adlink.handler.BroadlinkRemoteHandler] - rm3:c8b2c2c2[?]: updateItemStatus: Online -> Offline
```
When replacing the `clearProperty `again with `setProperty(bla,null)` it works...I can't tell you anythign else :)

**Comment by ynuyasha** [comment]
I reverted to `setProperty(bla,null)` and added `setProperty(String propName, @Nullable String propValue) {`
But build failed with this error
```
Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:compile (default-compile) on project org.openhab.binding.broadlink: Compilation failure
[ERROR] openhab2-addons\openhab2-addons\bundles\org.openhab.binding.broadlink\src\main\java\org\openhab\binding\broadlink\handler\BroadlinkBaseThingHandler.java:[298,11772] Null type mismatch (type annotations): required '@NonNull String' but this expression has type '@Nullable String'
```

this is my env:
```
mvn -v
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: C:\blabla\apache-maven-3.6.3\bin\..
Java version: 1.8.0_242, vendor: AdoptOpenJDK, runtime: C:\Program Files\AdoptOpenJDK\jdk-8.0.242.08-hotspot\jre
Default locale: it_IT, platform encoding: Cp1252
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"
```

**Comment by tknaller** [comment]
this is my env:
```
$ mvn -v
Apache Maven 3.6.0
Maven home: /usr/share/maven
Java version: 11.0.6, vendor: Ubuntu, runtime: /usr/lib/jvm/java-11-openjdk-amd64
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "4.4.59+", arch: "amd64", family: "unix"
```

apparantly my ubuntu has multiple JDKs and is using ver 11:
```
$ dpkg --list | grep jdk
ii  openjdk-11-jre-headless:amd64         11.0.6+10-1ubuntu1~18.04.1          amd64        OpenJDK Java runtime, using Hotspot JIT (headless)
ii  openjdk-8-jdk:amd64                   8u242-b08-0ubuntu3~18.04            amd64        OpenJDK Development Kit (JDK)
ii  openjdk-8-jdk-headless:amd64          8u242-b08-0ubuntu3~18.04            amd64        OpenJDK Development Kit (JDK) (headless)
ii  openjdk-8-jre:amd64                   8u242-b08-0ubuntu3~18.04            amd64        OpenJDK Java runtime, using Hotspot JIT
ii  openjdk-8-jre-headless:amd64          8u242-b08-0ubuntu3~18.04            amd64        OpenJDK Java runtime, using Hotspot JIT (headless)
``` 

that could be the difference :)

**Comment by ynuyasha** [comment]
i set maven to use java 11 but the result is the same on both java 8 or 11

```
mvn -v
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: apache-maven-3.6.3\bin\..
Java version: 11.0.6, vendor: AdoptOpenJDK, runtime: C:\Program Files\AdoptOpenJDK\jdk-11.0.6.10-hotspot
Default locale: it_IT, platform encoding: Cp1252
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"
```

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:compile (default-compile) on project org.openhab.binding.broadlink: Compilation failure
[ERROR] openhab2-addons\openhab2-addons\bundles\org.openhab.binding.broadlink\src\main\java\org\openhab\binding\broadlink\handler\BroadlinkBaseThingHandler.java:[298,11772] Null type mismatch (type annotations): required '@NonNull String' but this expression has type '@Nullable String'
```

**Comment by tknaller** [comment]
haha awesome that goes beyond my java knowledge 😄  sorry

File: bundles/org.openhab.binding.broadlink/src/main/java/org/openhab/binding/broadlink/handler/BroadlinkBaseThingHandler.java [hunk 1]
```diff
@@ -166,16 +170,20 @@ protected boolean authenticate() {
...
             id = new byte[4];
+            Arrays.fill(id, (byte)0);
         } else {
             id = Hex.fromHexString(properties.get("id"));
         }
         byte key[];
-        if (properties.get("key") == null || properties.get("id") == null) {
+        if (StringUtils.isEmpty(properties.get("key"))) {
+            
             key = Hex.convertHexToBytes(thingConfig.getAuthorizationKey());
```
Comment (ynuyasha) [inline bundles/org.openhab.binding.broadlink/src/main/java/org/openhab/binding/broadlink/handler/BroadlinkBaseThingHandler.java]: i found a strange behavior often key is empty

**Comment by ynuyasha** [comment]
i have rollback to setproperty and add some trace logs. Can you try and tell me something about key variable too?

File: bundles/org.openhab.binding.broadlink/src/main/java/org/openhab/binding/broadlink/handler/BroadlinkBaseThingHandler.java (hunk 1 above)
Comment (ynuyasha) [reply bundles/org.openhab.binding.broadlink/src/main/java/org/openhab/binding/broadlink/handler/BroadlinkBaseThingHandler.java]: if i restart openhab key was get correctly from thing

**Comment by tknaller** [comment]
yeah that works 👍

**Pull request merged**